zrml/neo-swaps/src/tests/deploy_combinatorial_pool.rs

```diff
@@ -0,0 +1,536 @@
+        let spot_prices = vec![BASE / (asset_count as u128); asset_count];
+        let swap_fee = CENT;
+        let (market_ids, pool_id) = create_markets_and_deploy_combinatorial_pool(
+            ALICE,
+            BASE_ASSET,
+            vec![
+                MarketType::Categorical(2),
+                MarketType::Categorical(4),
+                MarketType::Scalar(0u128..=1u128),
+            ],
+            amount,
+            spot_prices.clone(),
+            swap_fee,
+        );
+        let pool = Pools::<Runtime>::get(pool_id).unwrap();
+        let assets = pool.assets();
+        let expected_liquidity = 36_067_376_022;
+        let buffer = AssetManager::minimum_balance(pool.collateral);
+        assert_eq!(pool.assets(), assets);
+        assert_approx!(pool.liquidity_parameter, expected_liquidity, 1);
+        assert_eq!(pool.collateral, BASE_ASSET);
+        assert_liquidity_tree_state!(
+            pool.liquidity_shares_manager,
+            [Node::<Runtime> {
+                account: Some(ALICE),
+                stake: amount,
+                fees: 0u128,
+                descendant_stake: 0u128,
+                lazy_fees: 0u128,
+            }],
+            create_b_tree_map!({ ALICE => 0 }),
+            Vec::<u32>::new(),
+        );
+        assert_eq!(pool.swap_fee, swap_fee);
+        assert_balance!(pool.account_id, pool.collateral, buffer);
+
+        let mut reserves = BTreeMap::new();
+        for (&asset, &price) in assets.iter().zip(spot_prices.iter()) {
+            assert_balance!(pool.account_id, asset, amount);
+            assert_eq!(pool.reserve_of(&asset).unwrap(), amount);
+            assert_eq!(pool.calculate_spot_price(asset).unwrap(), price);
+            assert_balance!(ALICE, asset, 0);
+            reserves.insert(asset, amount);
+        }
+        assert_balance!(ALICE, BASE_ASSET, alice_before - amount - buffer);
+
+        System::assert_last_event(
+            Event::CombinatorialPoolDeployed {
+                who: ALICE,
+                market_ids,
+                pool_id,
+                account_id: pool.account_id,
+                reserves,
+                collateral: pool.collateral,
+                liquidity_parameter: pool.liquidity_parameter,
+                pool_shares_amount: amount,
+                swap_fee,
+            }
+            .into(),
+        );
+    });
+}
+
+#[test]
+fn deploy_combinatorial_pool_fails_on_incorrect_vec_len() {
+    ExtBuilder::default().build().execute_with(|| {
+        let market_ids = vec![
+            create_market(ALICE, BASE_ASSET, MarketType::Scalar(0..=1), ScoringRule::AmmCdaHybrid),
+            create_market(ALICE, BASE_ASSET, MarketType::Categorical(3), ScoringRule::AmmCdaHybrid),
+        ];
+        assert_noop!(
+            NeoSwaps::deploy_combinatorial_pool(
+                RuntimeOrigin::signed(ALICE),
+                6,
+                market_ids,
+                _10,
+                vec![20 * CENT; 5],
```

Comment:
Comment here that the above markets produce `2 * 3 = 6` collections, instead of `5`: `LONG & 0`, `LONG & 1`, `LONG & 2`, `SHORT & 0`, `SHORT & 1`, `SHORT & 2`.

Reply:
4d0078d62168b62ac02f61cfe335adfb0b492079